### PR TITLE
6 fix constructors in parser

### PR DESCRIPTION
**Pull request by Lubow-k**
* fill_args deleted, now this logic in constructors 
* Add special constructor for ParserException
* Corrections in simple tests

### CODE REVIEW

File: instructions/instructions_impl.cpp [hunk 1]
```diff
@@ -4,87 +4,123 @@
...
-    throw ParserException(
-        Parser::exception_message("add", args_amount, args.size()));
+    throw ParserException("li", args_amount, args.size());
+  }
+  Register dist_ = Parser::get_register(args[0]);
+  int immediate_;
+  if (Parser::is_number(args[1])) {
+      immediate_ = stoi(args[1]);
+  } else {
+    throw ParserException("invalid immediate in li: " + args[1]);
```
Comment (rustam2027) [inline instructions/instructions_impl.cpp:47]: You change ParserException constructor right?

File: instructions/instructions_impl.cpp [hunk 2]
```diff
@@ -4,87 +4,123 @@
...
-  dist = Parser::get_register(args[0]);
-  source = Parser::get_register(args[1]);
+  Register dist_ = Parser::get_register(args[0]);
+  Register source_ = Parser::get_register(args[1]);
+  int immediate_;
   if (Parser::is_number(args[2])) {
-      immediate = stoi(args[2]);
+      immediate_ = stoi(args[2]);
   } else {
     throw ParserException("invalid immediate in addi: " + args[2]);
```
Comment (rustam2027) [inline instructions/instructions_impl.cpp:74]: Здесь тоже

File: instructions/instructions_impl.cpp [hunk 3]
```diff
@@ -94,15 +130,20 @@ void Mv::exec(State &state) {
...
   }
-  dist = Parser::get_register(args[0]);
-  source = Parser::get_register(args[1]);
+  Register dist_ = Parser::get_register(args[0]);
+  Register source1_ = Parser::get_register(args[1]);
+  Register source2_ = Parser::get_register(args[2]);
+
+  dist = dist_;
+  source1 = source1_;
+  source2 = source2_;
```
Comment (rustam2027) [inline instructions/instructions_impl.cpp:146]: Why use this buffers

**Review by rustam2027 — CHANGES_REQUESTED** [review]

File: instructions/instructions_impl.cpp (hunk 1 above)
Comment (Lubow-k) [reply instructions/instructions_impl.cpp:47]: Yes, I moved the function from Parser to the ParserException constructor. I wrote about all the changes in the description of pr, you can check it)

File: instructions/instructions_impl.cpp (hunk 3 above)
Comment (Lubow-k) [reply instructions/instructions_impl.cpp:146]: To avoid partial filling of fields, these functions (like get_register) can throw exceptions

**Pull request merged**